library/std/src/env.rs

```diff
@@ -185,15 +185,9 @@ impl fmt::Debug for VarsOs {
 ///
 /// # Errors
 ///
-/// Errors if the environment variable is not present.
-/// Errors if the environment variable is not valid Unicode. If this is not desired, consider using
-/// [`var_os`].
-///
-/// # Panics
-///
-/// This function may panic if `key` is empty, contains an ASCII equals sign
-/// `'='` or the NUL character `'\0'`, or when the value contains the NUL
-/// character.
```

Comment:
Wouldn't it be better to still say what happens for these cases?

Reply:
The error section (which you commented on) should cover it? The theory is that "if the environment variable isn't set" applies to the equals sign and NUL character being in the key, since we should only be erroring on platforms where those *can't* be in the key. The value containing the NUL character is explicitly covered.

Reply:
> The theory is that "if the environment variable isn't set" applies to the equals sign and NUL character being in the key, since we should only be erroring on platforms where those can't be in the key.

That seems awfully implicit, doesn't it? I don't have a strong opinion, but given in particular that behavior changed here, I think it is worth saying explicitly that empty, `=` or NUL are considered "not set".

> The value containing the NUL character is explicitly covered.

Is it? I don't see it. It talks about invalid unicode, but NULL is not invalid unicode AFAIK.

Reply:
> That seems awfully implicit, doesn't it? I don't have a strong opinion, but given in particular that behavior changed here, I think it is worth saying explicitly that empty, = or NUL are considered "not set".

The thing is that they're not always considered not set. They're considered not set (hopefully, if I understand things right) on platforms where they could never have been set in the first place. There are platforms on which it is valid to have an = or a NUL. At that point, I'd basically be saying "If the environment variable isn't set (including if it isn't set because it contains an `=` or NUL character and that is invalid for the current platform)". Of course, I could say it more nicely than that, but it doesn't seem like the reason why we think environment variable isn't set should matter?

> Is it? I don't see it. It talks about invalid unicode, but NULL is not invalid unicode AFAIK.

Uhhh. Whoops. Will fix.

Reply:
> At that point, I'd basically be saying "If the environment variable isn't set (including if it isn't set because it contains an = or NUL character and that is invalid for the current platform)". Of course, I could say it more nicely than that, but it doesn't seem like the reason why we think the environment variable isn't set should matter?

I think this is worth saying explicitly, given in particular that future Rust versions will behave different from older Rust versions here. I am sure some people will wonder what happens in this case, so the text should answer their question. The docs are very short currently so I am not worried about adding a sentence or two.

Btw, looking at https://doc.rust-lang.org/nightly/std/env/fn.var.html the text renders this as `[None]`, which also doesn't seem right -- we don't want those brackets to show.

Reply:
I think I flipped the brackets and the quotes. *sighs*